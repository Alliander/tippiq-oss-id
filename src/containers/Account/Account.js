/**
 * Account container.
 * @module containers/Account
 */

import React, { PropTypes } from 'react';
import Helmet from 'react-helmet';
import { asyncConnect } from 'redux-connect';
import { Grid, Row, Col } from 'react-bootstrap';

import { getAppConfig } from '../../actions';
import { AccountMenu } from '../../components';
import styles from './Account.scss';

/**
 * This class defines the Account container.
 * @function Account
 * @param {Object} props Component properties.
 * @param {Object} props.children Child nodes.
 * @returns {string} Markup for the component.
 */
export const Account = ({ children }) =>
  <div className={styles.container}>
    <Helmet title="Mijn Account" />
    <Grid>
      <Row>
        <Col xs={12} sm={3}>
          <AccountMenu />
        </Col>
        <Col xs={12} sm={9} className={styles.mainContent}>
          {children}
        </Col>
      </Row>
    </Grid>
  </div>;

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
Account.propTypes = {
  children: PropTypes.object.isRequired,
};

export default
asyncConnect([{
  key: 'appConfig',
  promise: ({ store: { dispatch } }) => dispatch(getAppConfig()),
}])(Account);
