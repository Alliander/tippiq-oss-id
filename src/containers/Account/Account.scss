@import "../../theme/variables/colors.css";

.container{
  margin: 20px 0 100px 0;
}

.mainContent{
  -webkit-flex: 2 0 0;
  flex: 2 0 0;
  background-color:  var(--whiteColor);;
  padding: 20px;
  box-shadow: 0px 0px 4px rgba(0, 0, 0, .3)
}
