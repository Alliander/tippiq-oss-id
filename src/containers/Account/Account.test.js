import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Account } from './Account';

test('Basic render snapshot test', () => {
  const component = shallow(<Account><div>content</div></Account>);

  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});

