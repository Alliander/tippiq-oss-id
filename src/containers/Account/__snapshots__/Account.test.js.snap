// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic render snapshot test 1`] = `
<div>
  <HelmetWrapper
    title="Mijn Account"
  />
  <Grid
    bsClass="container"
    componentClass="div"
    fluid={false}
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        sm={3}
        xs={12}
      >
        <AccountMenu />
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        sm={9}
        xs={12}
      >
        <div>
          content
        </div>
      </Col>
    </Row>
  </Grid>
</div>
`;
